### PR TITLE
add zkdx adapter

### PR DESCRIPTION
**Pull request by helgaxdev**

### CODE REVIEW

**Comment by nitish-91** [comment]
@helgaxdev  please create a branch from the main branch of this repo, make your changes in that branch and create pull request to main. 
https://github.com/delta-hq/l2-lxp-liquidity-reward#notes

**Comment by nitish-91** [comment]
> @helgaxdev please create a branch from the main branch of this repo, make your changes in that branch and create pull request to main. https://github.com/delta-hq/l2-lxp-liquidity-reward#notes

Please ignore this comment, we will figure this out internally

File: adapters/zkdx/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+            })
+        {
+            user_address
+            token_address
+            token_balance
+            block_number
+        }
+    }`
+
+export const getUserTVLByBlock = async (block_number: number) => {
```
Comment (nitish-91) [inline adapters/zkdx/src/index.ts:26]: @helgaxdev please check the readme and make sure the input, output and Function name matches the expectation,

File: adapters/zkdx/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+const lineaProvider = new ethers.providers.JsonRpcProvider("https://rpc.linea.build");
+
+const getBalanceShots = gql`
+    query getBalanceShots($block_number: Int) {
+        tokenBalanceShots(
+            first: 1000
+            orderBy: block_number
+            orderDirection: desc
+            where: {
+                block_number_lte: $block_number
```
Comment (0xroll) [inline adapters/zkdx/src/index.ts:16]: we will need pagination support if there are > 1000 users

File: adapters/zkdx/src/index.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        let timestampStr = new Date(timestamp * 1000).toISOString();
+        console.log(`Snapshot for block: ${block}:`);
+        let balanceMap = await getUserTVLByBlock(block);
+
+        for (let [user_address, tokenBalances] of balanceMap) {
+            for (let [token_address, token_balance] of tokenBalances) {
+                if (BigNumber.from(token_balance).lt(100))
+                    continue;
+                let token_symbol = token_address == "0x176211869ca2b568f2a7d4ee941e073a821ee1ff" ? "USDC" : "WETH";
+                let units = token_symbol == "USDC" ? 6 : 18;
```
Comment (0xroll) [inline adapters/zkdx/src/index.ts:84]: are there just 2 assets? might be better to have a mapping table

File: adapters/zkdx/src/index.ts (hunk 1 above)
Comment (helgaxdev) [reply adapters/zkdx/src/index.ts]: fixed

File: adapters/zkdx/src/index.ts (hunk 3 above)
Comment (helgaxdev) [reply adapters/zkdx/src/index.ts]: 2 assets for now, fixed

**Pull request closed without merging**